File: connectors/connector-msp/src/test/java/com/taskadapter/connector/msp/MSTaskLoaderIntegrationTest.java
package com.taskadapter.connector.msp;

import com.taskadapter.connector.testlib.TestUtils;
import com.taskadapter.model.AllFields;
import com.taskadapter.model.CustomString;
import net.sf.mpxj.TaskField;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MSTaskLoaderIntegrationTest {
    @Test
    public void testLoadFileCreatedByMSPWith1Task() {
        var tasks = MSPTestUtils.load("created_by_msp_1task.xml");
        assertThat(tasks).hasSize(1);
    }

    @Test
    public void testFind1Task() {
        var tasks = MSPTestUtils.load("created_by_msp_1task.xml");
        var myTaskAddedFromMSP = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "task1");
        assertThat(myTaskAddedFromMSP)
                .withFailMessage("required task not found in the tasks list")
                .isNotNull();
    }

    @Test
    public void testLoadFileCreatedByMSPWithManyTasks() {
        var tasks = MSPTestUtils.load("created_by_msp_tasks.xml");
        assertThat(tasks).hasSize(4);

        TestUtils.findTaskByFieldNameOrFail(tasks, AllFields.summary, "task1");
        TestUtils.findTaskByFieldNameOrFail(tasks, AllFields.summary, "task1-sub1");
        TestUtils.findTaskByFieldNameOrFail(tasks, AllFields.summary, "task2");
    }

    @Test
    public void loadFileCreatedByTaskAdapter() {
        var tasks = MSPTestUtils.load("created_by_ta_27.xml");
        assertThat(tasks).hasSize(27);
        var t1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "improve components");
        assertThat(t1).withFailMessage("required task not found in the tasks list")
                .isNotNull();

        // verify TextXX fields are loaded. this is a regression test for
        // https://bitbucket.org/taskadapter/taskadapter/issues/74/issue-type-is-reset-to-default-setting
        assertThat(t1.getValue(new CustomString(TaskField.TEXT1.getName())))
                .isEqualTo("49");
        assertThat(t1.getValue(new CustomString(TaskField.TEXT2.getName())))
                .isEqualTo("Feature");

        var sub1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "sub1");
        assertThat(sub1)
                .withFailMessage("required task not found in the tasks list")
                .isNotNull();
    }

    @Test
    public void loadFileCreatedByTA1Task() {
        var tasks = MSPTestUtils.load("created_by_ta_1.xml");
        assertThat(tasks).hasSize(1);
        var t1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "support me!");
        assertThat(t1)
                .withFailMessage("required task not found in the tasks list")
                .isNotNull();
    }

    @Test
    public void emptyLinesInMspXmlFileAreSkipped() {
        // total number of lines is 170 with 163 non-empty ones
        var tasks = MSPTestUtils.load("IT_Department_Project_Master.xml");
        assertThat(tasks).hasSize(163);
    }
}

File: connectors/connector-msp/src/test/scala/com/taskadapter/connector/msp/MSTaskLoaderIntegrationTest.scala
package com.taskadapter.connector.msp

import com.taskadapter.connector.testlib.TestUtils
import com.taskadapter.model.{AllFields, CustomString, Summary}
import net.sf.mpxj.TaskField
import org.junit.Assert
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

import scala.collection.JavaConverters._

@RunWith(classOf[JUnitRunner])
class MSTaskLoaderIntegrationTest extends FunSpec with Matchers {
  it("testLoadFileCreatedByMSPWith1Task") {
    val tasks = MSPTestUtils.load("created_by_msp_1task.xml")
    Assert.assertEquals(1, tasks.size)
  }

  it("testFind1Task") {
    val tasks = MSPTestUtils.load("created_by_msp_1task.xml")
    val myTaskAddedFromMSP = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "task1")
    if (myTaskAddedFromMSP == null) Assert.fail("required task not found in the tasks list")
  }

  it("testLoadFileCreatedByMSPWithManyTasks") {
    val tasks = MSPTestUtils.load("created_by_msp_tasks.xml")
    Assert.assertEquals(4, tasks.size)
    val t1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "task1")
    Assert.assertNotNull("required task not found in the tasks list", t1)
    val t1Sub1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "task1-sub1")
    Assert.assertNotNull("required task not found in the tasks list", t1Sub1)
    val t2 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "task2")
    Assert.assertNotNull("required task not found in the tasks list", t2)
  }

  it("load file created by TA") {
    val tasks = MSPTestUtils.load("created_by_ta_27.xml")
    Assert.assertEquals(27, tasks.size)
    val t1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "improve components")
    Assert.assertNotNull("required task not found in the tasks list", t1)

    // verify TextXX fields are loaded. this is a regression test for
    // https://bitbucket.org/taskadapter/taskadapter/issues/74/issue-type-is-reset-to-default-setting
    t1.getValue(new CustomString(TaskField.TEXT1.getName)) shouldBe "49"
    t1.getValue(new CustomString(TaskField.TEXT2.getName)) shouldBe "Feature"

    val sub1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "sub1")
    Assert.assertNotNull("required task not found in the tasks list", sub1)
  }

  it("load file created by TA 1 task") {
    val tasks = MSPTestUtils.load("created_by_ta_1.xml")
    Assert.assertEquals(1, tasks.size)
    val t1 = TestUtils.findTaskByFieldName(tasks, AllFields.summary, "support me!")
    Assert.assertNotNull("required task not found in the tasks list", t1)
  }

  it("empty lines in MSP XML file are skipped") {
    // total number of lines is 170 with 163 non-empty ones
    val tasks = MSPTestUtils.load("IT_Department_Project_Master.xml").asScala
    Assert.assertEquals(163, tasks.size)
  }
}
